2005-11-08  H.J. Lu  <hongjiu.lu at intel.com>

	* sysdeps/wordsize-32/Makefile (CPPFLAGS-divdi3.c): Set to
	-Din_divdi3_c.
	* sysdeps/wordsize-32/symbol-hacks.h: Skip if in_divdi3_c
	is defined.

--- sysdeps/wordsize-32/Makefile	19 Sep 2002 06:47:56 -0000	1.5
+++ sysdeps/wordsize-32/Makefile	16 Nov 2005 19:19:03 -0000	1.6
@@ -2,5 +2,6 @@ ifeq ($(subdir),csu)
 ifeq (yes,$(build-shared))
 sysdep_routines += divdi3
 shared-only-routines += divdi3
+CPPFLAGS-divdi3.c = -Din_divdi3_c
 endif
 endif
--- sysdeps/wordsize-32/symbol-hacks.h	8 Mar 2004 20:59:34 -0000	1.1
+++ sysdeps/wordsize-32/symbol-hacks.h	16 Nov 2005 19:18:25 -0000	1.2
@@ -22,7 +22,7 @@
    therefore we get PLTs.  Unnecessarily so.  Changing gcc is a big
    task which might not be worth it so we play tricks with the
    assembler.  */
-#if !defined __ASSEMBLER__ && !defined NOT_IN_libc && defined SHARED
+#if !defined __ASSEMBLER__ && !defined in_divdi3_c && !defined NOT_IN_libc && defined SHARED
 asm ("__divdi3 = __divdi3_internal");
 asm ("__udivdi3 = __udivdi3_internal");
 asm ("__moddi3 = __moddi3_internal");
